Adiciona footer

jppaulo06 opened the pull request.

jppaulo06 committed:
Cria o Footer. Responsividade ainda ruim.

jppaulo06 committed:
Merge branch 'main' of github.com:uspcodelab/bootcamp-2022 into main

josemayer committed:
Integra main na branch joao-footer

josemayer reviewed (commented):
Muito bom, JP!!!

Algumas observações que considero importante ressaltar:
- O `v-container` do _footer_ não está alinhado com o `v-container` das páginas, porque você definiu um _padding_ de 0 para ele e uma largura fixa de 100%.
- A tag `p` não é muito adequada para se "quebrar" linhas. Uma forma boa de se fazer isso é definir um componente flex de direção coluna e encaixar algumas tags `span` dentro dele (cada uma é um item do flex e, portanto, tem uma linha independente).
- Os _breakpoints_  estão um pouco confusos. A melhor forma de defini-los é categorizar a quantidade de colunas como na menor tela e, a partir disso, para telas maiores, definir as tags correspondentes. (Ex: se eu defino uma coluna que ocupa 12 linhas no celular e 4 nas telas médias ou maiores, ela ficaria da seguinte forma: `<v-col cols="12" md="4"></v-col>`. Note que o atributo md indica a quantidade de unidades de coluna para telas md ou maiores).
- Você definiu um atributo **col** para as tags `v-col`, mas o atributo é **cols**.
- Não precisamos nos preocupar com a altura do _footer_. Se ela ficar sem essa propriedade, ela recebe seu valor automático, que é definido com base no conteúdo que ela tem dentro. Isso permite que ele se estenda em telas menores para conter todo o conteúdo.

Tenho, agora, algumas sugestões de melhora acessórias para o código:
- O vuetify dispõe de alguns _flex helpers_ que ajudam bastante na hora de criar classes que precisam apenas cumprir algum papel como flex. Recomendo dar uma olhada neles [aqui](https://vuetifyjs.com/en/styles/flex/#flex-direction) (eles têm até flex com _breakpoint_, é legalzão e facilita muito a vida na hora de fazer as coisas serem responsivas 😁). Isso pouparia escrever muitas classes nos estilos.
- Ao invés de utilizar imagens, você pode utilizar ícones do próprio vuetify com a tag `v-icon`, que, por padrão, utiliza os ícones da **Material Design Icons**. O interessante de utilizá-los é que eles se comportam como um svg e, consequentemente, como um texto, sendo personalizáveis (podemos trocar sua cor, adicionar sombras, bordas, etc.). [Aqui](https://vuetifyjs.com/en/components/icons/) tem algumas informações a mais de uso.

josemayer committed:
Atualiza footer para realizar a mudança automática de altura

Signed-off-by: Jose Lucas Silva Mayer <josemayer@usp.br>

josemayer committed:
Remove imagens de ícones e move logo do IME para a página de assets

Signed-off-by: Jose Lucas Silva Mayer <josemayer@usp.br>

josemayer commented:
Outra sugestão que esqueci de ressaltar:
- Idealmente, devemos guardar as imagens necessárias na pasta **assets**, pois isso centraliza a organização dos códigos e repositório e facilita a recuperação desses conteúdos.

josemayer committed:
Atualiza caminho da imagem da logo para a pasta de assets

Signed-off-by: Jose Lucas Silva Mayer <josemayer@usp.br>

josemayer committed:
Corrige inconsistências de grade e incorpora mudanças sugeridas

Remove incoerência do v-container com o restante das páginas, adiciona a tag v-row, corrige breakpoints, corrige ícones,
adiciona flex-helpers para gerenciar as classes flex e faz mudanças pontuais nos textos e formatação.

Signed-off-by: Jose Lucas Silva Mayer <josemayer@usp.br>

josemayer commented:
Uma última coisa que é um erro que eu não citei:
- As tags `v-col` devem sempre estar dentro de tags `v-row` para funcionarem normalmente. Elas, por si só, garantem um sistema de _grid_ flex. [Essa página](https://v15.vuetifyjs.com/en/framework/grid/) tem exemplos bons desse sistema.

josemayer committed:
Corrige erros de linter

Signed-off-by: Jose Lucas Silva Mayer <josemayer@usp.br>

josemayer commented:
Fiz algumas mudanças para corrigir os erros que citei e incorporar as sugestões:
- Removi o id do `v-container`, para que o _footer_ fique na mesma configuração das páginas e esteja dentro da "margem" de conteúdo.
- Adicionei uma tag `v-row` que incorpora as tags `v-col`.
- Fiz as colunas ocuparem todas as unidades de colunas por padrão (preencherem uma linha) e nas telas médias ou maiores ocuparem apenas 4 unidades de colunas (formarem 3 colunas lado a lado). Também tirei as classes de flex delas, porque são redundantes: o sistema de grid já tem uma configuração flex.
- Removi as mudanças no _layout_ padrão, para que o _footer_ auto-ajuste sua altura de acordo com o conteúdo.
- Adicionei componentes `v-icon` nos lugares das imagens.
- Adicionei os _flex helpers_ para diminuir a quantidade de classes com diferentes propriedades flex no css. Como não existe nenhum _helper_ para a propriedade `justify-content: space-evenly`, deixei a classe "DivFooter-1" no código, para manter essa propriedade.

josemayer reviewed (approved):
Com as mudanças, LGTM!
